Project init

osyvokon opened the pull request.

osyvokon committed:
Project init

osyvokon committed:
Compile dictionary

osyvokon committed:
Find stress in non-ambigous cases

osyvokon committed:
Match parsed morphological tags to dictionary entries

This allows to resolve ambiguity based on case, part of speech and
singal/plular number.

osyvokon committed:
Configurable stress symbol

osyvokon committed:
Ignore case when looking up a word

osyvokon committed:
Preserve orginial whitespace

osyvokon committed:
Compress tags to bytes

osyvokon committed:
Add command line utility

osyvokon committed:
Include compiled dictionary as a package resource

osyvokon committed:
Rename package to ukrainian_word_stress

osyvokon committed:
Verbose mode and debug outputs

osyvokon committed:
Skip entries with wrong stress

osyvokon committed:
Configurable ambiguity resolution strategies

osyvokon committed:
Docs

osyvokon committed:
CLI option for printing version

osyvokon committed:
Add section on debugging and reporting issues

osyvokon committed:
Add a compiled dictionary

osyvokon committed:
Add example to README

dchaplinsky commented on README.md:
```diff
@@ -0,0 +1,143 @@
+* а́тлас - збірник карт
+* атла́с - тканина
+
+Resolving this is much harder and sometimes impossible.
```
Well, word frequency might give some improvement here, but I don't know where to obtain that. Maybe just ask lexicographs to put priorities (for example I saw вИгода much more often than вигОда)?

osyvokon replied:
(Discussed that in a call. Agreed to collect and label a corpus for heteronyms disambiguation)

dchaplinsky commented on ukrainian_word_stress/compile_dict.py:
```diff
@@ -0,0 +1,164 @@
+def _parse_dictionary(csv_path):
+    by_basic = collections.defaultdict(list)  # TODO: change to set
+    skipped = 0
+    for row in tqdm.tqdm(csv.DictReader(open(csv_path))):
```
You probably want to use this
https://github.com/tqdm/tqdm#redirecting-logging

osyvokon replied:
I just removed the logging under tqdm :)

dchaplinsky commented on ukrainian_word_stress/compile_dict.py:
```diff
@@ -0,0 +1,164 @@
+
+def count_vowels(s):
+    result = 0
+    for x in VOWELS:
```
return sum(s.count(x) for x in VOWELS) if you feel fancy.

dchaplinsky commented on ukrainian_word_stress/stressify_.py:
```diff
@@ -0,0 +1,210 @@
+
+    """
+
+    if not hasattr(stressify, "f"):
```
Maybe something like this:
```python

class Singleton(type):
    _instances: dict = {}

    def __call__(cls, *args, **kwargs):
        if cls not in cls._instances:
            cls._instances[cls] = super().__call__(*args, **kwargs)
        return cls._instances[cls]
```

osyvokon replied:
Thinking more about it, having a function that holds class instance(s) under the hood was not a good idea, considering that the class is quite heavy. To make it more transparent, I removed the function altogether.

dchaplinsky commented on ukrainian_word_stress/stressify_.py:
```diff
@@ -0,0 +1,210 @@
+            break
+    else:
+        # non-dictionary word
+        log.debug("%s is not in the dictionary", base)
```
you can still accent it, if it has just one vowel, no?

osyvokon replied:
Right. But since we don't show accent for single-syllable words, the end result will be the same.

dchaplinsky commented on ukrainian_word_stress/stressify_.py:
```diff
@@ -0,0 +1,210 @@
+        matches = accents_by_tags
+
+    log.info(matches)
+    log.info(on_ambiguity)
```
Do we need to keep those as log.info?

osyvokon replied:
Good catch!

osyvokon committed:
Fix incorrect multiple accents positions

dchaplinsky commented on setup.py:
```diff
@@ -0,0 +1,58 @@
+        ]
+    },
+    include_package_data=True,
+    install_requires=[
```
It's better to read those from requirements.txt

Also, add extra requirements for the building of the dictionary (TQDM?)

osyvokon replied:
Added tqdm to dev requirements. As of reading from requirements.txt -- with just two dependencies, it's an overkill, IMO.

osyvokon committed:
Add docs section on variative stress

osyvokon committed:
Add speed benchmark

osyvokon committed:
Include dev requirement for tqdm and other minor fixes

osyvokon committed:
Remove the `stressify` function.

a. It was not as configurable as the class
b. It was secretly holding a Stressifier object, which is quite heavy

osyvokon committed:
Formatting update

osyvokon committed:
Add benchmarking script
